lib/matplotlib/backends/backend_cairo.py

```diff
@@ -275,11 +273,7 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
             ctx.set_font_size(size)
 
             if HAS_CAIRO_CFFI:
-                if not isinstance(s, six.text_type):
-                    s = six.text_type(s)
-            else:
-                if six.PY2 and isinstance(s, six.text_type):
-                    s = s.encode("utf-8")
+                s = str(s)
```

Comment:
attn @anntzer 

This logic does not seem identical to be, but I am not sure of the context.

Reply:
The `else` block is a no-op on Python 3, so it can be removed.  That leaves the former block, and we can simplify the condition by observing that `str(some_string)` is a no-op too, and therefore calling `s = str(s)` unconditionally.

Reply:
Pretty sure the call to str/six.text_types was only necessary because of unicode/str issues; now *s* should always be a str and we shoudn't even have to call `str(s)`.